java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -120,9 +135,11 @@ public ExecutionState restoreParent() {
   }
 
   // FIXME : Hideous hack for closeable to get "Ignored" variables
+  @Deprecated
   public List<State> getStatesOf(Symbol symbol) {
     List<State> states = Lists.newArrayList();
-    for (SymbolicValue value : Iterables.concat(reachableValues.get(symbol), unreachableValues.get(symbol))) {
+    // FIXME(merciesa): why iterating over unreachableValues ?
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)